modules/server/src/main/scala/scaladex/server/Server.scala

```diff
@@ -164,13 +166,16 @@ object Server extends LazyLogging {
       DocumentationRoutes.routes
     )
     import session.implicits._
-    val userFacingRoute: Route =
+    val userFacingRoute =
       optionalSession(refreshable, usingCookies) { userId =>
-        val user = userId.flatMap(session.getUser)
-        frontPage.route(user) ~ adminPages.route(user) ~ awesomePages.route(user) ~
-          redirectToNoTrailingSlashIfPresent(StatusCodes.MovedPermanently) {
-            projectPages.route(user) ~ searchPages.route(user)
-          }
+        val futureMaybeUser = userId.traverse(session.getUser).map(_.flatten)
+        val futureRoute = futureMaybeUser.map { maybeUser =>
+          frontPage.route(maybeUser) ~ adminPages.route(maybeUser) ~ awesomePages.route(maybeUser) ~
+            redirectToNoTrailingSlashIfPresent(StatusCodes.MovedPermanently) {
+              projectPages.route(maybeUser) ~ searchPages.route(maybeUser)
+            }
+        }
+        Await.result(futureRoute, Duration.Inf)
```

Comment:
The code above expects an `Option[UUID] => Route`, but there's currently no way I can think of to call `GithubSession#getUser`, which evaluates to a `Future[_]` and then chain it to a raw `Route` type... any ideas?

Reply:
The definition of `Route` in akka is:
```scala
type Route = RequestContext => Future[RouteResult]
```

Here we have a `Future[Route]` and we can transform it into a route with:
```scala
context => futureRoute.flatMap(_.apply(context))
```